### PR TITLE
solve problem with no-unused-vars

### PR DESCRIPTION
**Pull request by lrezende**
I've cloned the repo... npm install... npm run... just that...
Getting "error  'EventItem' is defined but never used  no-unused-vars" error...
This pull request will prevent this

### CODE REVIEW

**Comment by tomaszmwojcik** [comment]
Thanks, I came across the same issue and this fixed it for me. Just to clarify, I was following the lesson 6 from Vue 3 + Typescript course (https://www.vuemastery.com/courses/vue3-typescript/data-with-custom-types/), on the `06-begin` branch and after adding type assertion as suggested in the course (`event: {} as EventItem`) the project stopped compiling with 
``` 
error  in ./src/views/EventDetails.vue

Module Error (from ./node_modules/eslint-loader/index.js):

(...)/Real-World-Vue-3-TypeScript/src/views/EventDetails.vue
  12:10  error  'EventItem' is defined but never used  no-unused-vars
```
It turns out Eslint doesn't think that type X in `as X` is used so either the `no-unused-vars` rule must be turned off (as in this PR) or some other configuration tweak is needed in order to make it compile again.

**Comment by samwhosung** [comment]
@bencodezen Why is this? Can this PR get merged, it's really annoying.

File: package.json [hunk 1]
```diff
@@ -50,7 +50,9 @@
     "parserOptions": {
       "parser": "@typescript-eslint/parser"
     },
-    "rules": {}
+    "rules": {
+      "no-unused-vars": "off"
```
Comment (Esurnir) [inline package.json:54]: Maybe this should be
```json
"rules": {
  "no-unused-vars": "off",
  "@typescript-eslint/no-unused-vars": "error"
}
```
per [this stackoverflow question](https://stackoverflow.com/questions/55280555/typescript-eslint-eslint-plugin-error-route-is-defined-but-never-used-no-un)

**Comment by bencodezen** [comment]
Sorry for the delay on this everyone! Thanks @lrezende for the original PR. I'll get this stuff cleaned up asap!

**Pull request merged**